tests/functional/tests.py

```diff
@@ -1370,42 +1370,55 @@ def test_thread_safe(client, test_file, log_output):
                           args=(bucket_name, object_name, test_file))
             thrd.start()
             thrd.join()
-        # Create a wrapper method to capture get_object api's return
-        # value during threading by introducing a list object to hold
-        # the return value from each call. Without this, the return
-        # value of get_object api calls will be lost during threading.
-        def threaded_get_object(cli, bckt_name, obj_name, obj_data):
-            obj_data.append(cli.get_object(bckt_name, obj_name))
+
+        # A list of exceptions raised by get object
+        # called in multiple threads.
+        exceptions = []
+
+        # get_object_and_check downloads an object, stores it in a file
+        # and then calculate its checksum, if the checksum result is
+        # not expected, set equal_checksums to False
```

Comment:
> The last part of the comment here is inaccurate. get_object_and_check method actually raises an exception instead of setting equal_checksums to False.

right, will be fixed.